Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -76,14 +76,19 @@ class TargetLinter: TargetLinting {
     private func lintProductName(target: Target) -> [LintingIssue] {
         var allowed = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_")
 
-        let allowsDot = target.product == .app || target.product == .commandLineTool
-        if allowsDot {
-            allowed.formUnion(CharacterSet(charactersIn: "."))
+        let productsAllowingExtraCharacters: Set<Product> = [
+            .app,
+            .bundle,
+            .commandLineTool,
+        ]
+        let allowsExtraCharacters = productsAllowingExtraCharacters.contains(target.product)
+        if allowsExtraCharacters {
+            allowed.formUnion(CharacterSet(charactersIn: ".-"))
         }
 
         if target.productName.unicodeScalars.allSatisfy(allowed.contains) == false {
             let reason =
-                "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9)\(allowsDot ? ", period (.)" : ""), and underscore (_) characters."
+                "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9)\(allowsExtraCharacters ? ", period (.), hyphen (-)" : ""), and underscore (_) characters."
```

Comment:
This message is not true for products that are not included in `productsAllowingExtraCharacters`, is it?

Reply:
Period and hyphen are allowed in product name for app, bundle, commandLineTool and, I believe, many other types of targets but framework.
I can check all types where Xcode automatically replaces these characters with underscore.

Reply:
Well, I've checked almost all targets in Xcode and there's only one type where it transforms the product name by replacing dots and hyphens with underscores. It's Framework. When you add it, Xcode automatically sets its Product Name to `$(TARGET_NAME:c99extidentifier)` so that, for example, a target named `my.frame-work` gets Product Name as `my_frame_work`.
I've also tried to add such targets (`.staticFramework` and `.framework`) in a Tuist project and it works differently, replacing only hyphens: `My.Frame-work` -> `My.Frame_work`.